### PR TITLE
docs: update documentation on cherry picking releases

### PR DESCRIPTION
**Pull request by cdrage**
docs: update documentation on cherry picking releases

### What does this PR do?

Updates our release documentation on how to cherry pick PRs and spin up
another release.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

N/A

### How to test this PR?

N/A

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -37,19 +37,44 @@ Below is what a typical release week may look like:
...
 ## Test release before it is rolling out.
 
-The release is a pre-release, it means it is not yet the latest version, so no clients will automatically update to this version.
+The release is a **pre-release**, it means it is not yet the latest version, so no clients will automatically update to this version.
 
-It allows QE (and everyone else) to test these binaries before they go live or are in the package managers.
+Inform QE the **pre-release** is out! It allows QE (and everyone else) to downnload and test these binaries before they go live or are in the package managers.
 
-At this step, generates as well Flathub pull request so people can test the package from the builders of the pull request.
+At this step, it also generates a Flathub pull request so people can test the package from the builders of the pull request.
```
Comment (benoitf) [inline RELEASE.md]: not sure about the 'it' changes in 'it also generates' as it's a person that needs to do that for now

File: RELEASE.md [hunk 2]
```diff
@@ -37,19 +37,44 @@ Below is what a typical release week may look like:
...
+1. Create a branch **FROM THE RELEASE**. Example, 1.3.x of release 1.3.0.
+2. Create PR(s) with the fixes that merge into the 1.3.x branch
+3. Make sure all PR's are merged
+
+**Re-spin a release:**
+
+You'll need to create another release from the 1.3.x branch. This can be done by doing the following:
+1. Go to `Run workflow` in the [release steps](/RELEASE.md#releasing-on-github) again.
+2. **MAKE SURE** you specify that you want to use the `1.3.x` branch, NOT `main` under `Branch to use for the release`.
+3. Version to release should be `1.3.x`
```
Comment (benoitf) [inline RELEASE.md]:
```suggestion
3. Version to release should be `1.3.z` (like 1.3.1, 1.3.2, etc.)
```

File: RELEASE.md (hunk 1 above)
Comment (cdrage) [reply RELEASE.md]: does it not automatically generate it? or we do it manually?

Comment (benoitf) [reply RELEASE.md]: we do it manually now

**Comment by cdrage** [comment]
@benoitf Updated! I removed the note regarding flatpak and instead said to refer to the assets generated in the pre-release.

File: RELEASE.md (hunk 2 above)
Comment (benoitf) [reply RELEASE.md]: could you replace .x by .z

**Review by benoitf — APPROVED** [review]

**Pull request merged**